### DISCUSSION TITLE
How to use U8x8 for ST7920 Display Driver?

### DISCUSSION THREAD

**Discussion by imrulkk89** [olikraus/u8g2#1390] [Q&A]
Hi All, 
In my current project, I'm using ST7920 128x64 GLCD display which is driven by ATmega32 by Hardware SPI. At first, I tried with U8g2 library and successfully connect the display and show text and graphics. But, ATmega32 has only 2,048 bytes of SRAM. There is not much RAM space left for other development if I run the u8g2 library with a full buffer for ST7920 128x64 GLCD display. I only need to print text. So, I found U8x8 implementation can do that. 

But, I can't find any constructor to initialize U8x8 for ST7920 128x64 GLCD display. 

How can I use u8x8 for ST7920 128x64 GLCD display by HW SPI in plain C for ATmega32 chip?

**Comment by olikraus** [comment]
U8x8 is not supported by st7920. 
You could use u8g2 in page mode. This will require much lesser RAM.

**Reply by imrulkk89** [reply]
@olikraus Thanks for the reply. I tried to use `u8g2_Setup_st7920_s_128x64_1()` & `u8g2_Setup_st7920_s_128x64_2()` function. But, both funcitons give distortion in display. How can I fix that?

**Reply by olikraus** [reply]
> How can I fix that? 

Maybe you can post a picture and some code
